activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields.any?
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
+            Passing arguments to the select method when used with a block is deprecated and will be removed in future version of Rails.
+          WARNING
+        end
+
+        return super()
```

Comment:
Yes because we need to make sure that we are not passing arguments to `Enumerable#select`